### PR TITLE
feat(ingest/teradata): Teradata source

### PR DESCRIPTION
**Pull request by treff7es**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Comment by treff7es** [comment]
CleanShot 2023-10-12 at 10 56 00
CleanShot 2023-10-12 at 10 55 43
CleanShot 2023-10-12 at 10 55 33

File: metadata-ingestion/src/datahub/utilities/sqlglot_lineage.py [hunk 1]
```diff
@@ -482,6 +482,7 @@ def _column_level_lineage(  # noqa: C901
         # Our snowflake source lowercases column identifiers, so we are forced
         # to do fuzzy (case-insensitive) resolution instead of exact resolution.
         "snowflake",
+        "teradata",
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/utilities/sqlglot_lineage.py:485]: let's add a comment around this

**Review by hsheth2 — APPROVED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/sql/teradata.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+                )
+                self.urns = None
+        else:
+            self.schema_resolver = SchemaResolver(
+                platform=self.platform,
+                platform_instance=self.config.platform_instance,
+                graph=None,
+                env=self.config.env,
+            )
+            self.urns = None
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/ingestion/source/sql/teradata.py:168]: We shouldn't have to do this sort of logic to start. I think for now we should just always ingest schema metadata, pass that to the schema resolver, and never pass a graph into the schema resolver

**Review by asikowitz — CHANGES_REQUESTED** [review]
Can we add an image to the frontend for teradata. I think that's an important part of the experience (lol). I think the lineage query needs to be changed, and I think we should pass ingested schema metadata into the schema resolver

**Pull request merged**